drivers/pcie/host/pcie.c

```diff
@@ -126,34 +126,39 @@ static bool pcie_get_bar(pcie_bdf_t bdf,
 			 bool io)
 {
 	uint32_t reg = bar_index + PCIE_CONF_BAR0;
+	uint32_t cmd_reg = pcie_conf_read(bdf, PCIE_CONF_CMDSTAT);
+	bool ret = false;
 #ifdef CONFIG_PCIE_CONTROLLER
 	const struct device *dev;
 #endif
 	uintptr_t phys_addr;
 	size_t size;
 
+	/* IO/memory decode should be disabled before sizing/update BAR. */
+	pcie_conf_write(bdf, PCIE_CONF_CMDSTAT, cmd_reg & (~0x3));
+
 #ifdef CONFIG_PCIE_CONTROLLER
 	dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_pcie_controller));
 	if (!dev) {
 		LOG_ERR("Failed to get PCIe root complex");
-		return false;
+		goto err_exit;
 	}
 #endif
 
 	if (reg > PCIE_CONF_BAR5) {
-		return false;
+		goto err_exit;
 	}
```

Comment:
Wouldn't it make more sense to do the decode disabling only after these two branches, since they're just checking for valid function input parameters?

Reply:
updated